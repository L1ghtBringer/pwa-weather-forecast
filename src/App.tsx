import React from 'react'
import Forecast from 'pages/Forecast/Forecast'
import { AppWrapper, LoveMercury } from './App.styles'

const App: React.FC = () => {
  return <AppWrapper>
    <Forecast/>
    <LoveMercury>C ЛЮБОВЬЮ ОТ MERCURY DEVELOPMENT</LoveMercury>
  </AppWrapper>
}

export default App
