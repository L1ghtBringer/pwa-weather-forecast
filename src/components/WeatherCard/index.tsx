import React from 'react'
import { WeatherContainer } from './weathercatd.styles'
import { IWeatherDay } from '../../interfaces'

const WeatherCard: React.FC<IWeatherDay> = ({ time, temperature, icon }) => {
  const getDateString = (date: number): string => {
    const dateFormat = new Date(date * 1000)
    const month = dateFormat.toLocaleString('en', { month: 'long' })
    return `${dateFormat.getDate()} ${month} ${dateFormat.getFullYear()}`
  }

  return (
    <WeatherContainer>
      <span>{getDateString(time)}</span>
      <img src={require(`./icons/${icon}.png`).default} alt="icons" />
      <span>
        {temperature > 0
          ? `+${temperature.toFixed(0)}°`
          : `${temperature.toFixed(0)}°`}
      </span>
    </WeatherContainer>
  )
}

export default WeatherCard
