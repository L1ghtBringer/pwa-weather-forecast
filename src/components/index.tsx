export { default as Calendar } from './Calendar'
export { default as DatePicker } from './DatePicker'
export { default as Select } from './Select'
export { default as FillFields } from './FillFields'
export { default as WeatherCard } from './WeatherCard'
export { default as Carousel } from './Carousel'
export { default as Snackbar } from './Snackbar'
