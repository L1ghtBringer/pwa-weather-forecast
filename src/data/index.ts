export const cities = [
  {
    value: {
      latitude: 55.796127,
      longitude: 49.106405
    },
    label: 'Kazan'
  },
  {
    value: {
      latitude: 45.03547,
      longitude: 38.975313
    },
    label: 'Krasnodar'
  },
  {
    value: {
      latitude: 53.195873,
      longitude: 50.100193
    },
    label: 'Samara'
  },
  {
    value: {
      latitude: 51.533557,
      longitude: 46.034257
    },
    label: 'Saratov'
  },
  {
    value: {
      latitude: 53.507836,
      longitude: 49.420393
    },
    label: 'Tolyatti'
  }
]
