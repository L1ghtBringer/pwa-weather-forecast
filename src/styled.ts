import baseStyled, { ThemedStyledInterface } from 'styled-components'

export const theme = {
  media: {
    laptop: '(max-width: 1366px)',
    tablet: '(max-width: 805px)',
    tabletS: '(max-width: 700px)',
    phone: '(max-width: 520px)'
  },
  color: {
    accent: '#373AF5',
    secondary: {
      active: '#8083A4',
      hover: '#8083A499'
    },
    baseStong: '#2C2D76',
    critic: '#E5596D',
    baseWeak: '#FFFFFF'
  },
  font: 'Ubuntu, sans-serif'
}

export type Theme = typeof theme

export const styled = baseStyled as ThemedStyledInterface<Theme>
