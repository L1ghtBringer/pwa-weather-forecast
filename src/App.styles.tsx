import styled from 'styled-components'

export const AppWrapper = styled.div`
  width: 100%;
  min-height: 100vh;
  height: auto;
  position: relative;
`

export const LoveMercury = styled.p`
  color: #FFFFFF;
  opacity: 0.6;
  font-family: Ubuntu, sans-serif;
  font-size: 14px;
  position: absolute;
  left: calc(50% - 140px);
  bottom: 16px;
`
