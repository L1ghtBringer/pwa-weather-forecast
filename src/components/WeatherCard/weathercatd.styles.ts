import styled from 'styled-components'

export const WeatherContainer = styled.div`
  width: 100%;
  background: ${(props) => props.theme.color.accent};
  border: 2px solid ${(props) => props.theme.color.accent};
  box-shadow: 0px 4px 4px rgba(4, 5, 73, 0.25),
    14px 14px 20px rgba(5, 6, 114, 0.2);
  border-radius: 8px;
  padding: 18px;
  display: flex;
  flex-direction: column;
  user-select: none;
  span {
    font-size: 16px;
    line-height: 24px;
    text-transform: lowercase;
    font-weight: bold;
    color: ${(props) => props.theme.color.baseWeak};
    font-family: Ubuntu;
  }
  span:last-child {
    font-size: 32px;
    line-height: 32px;
    margin-left: auto;
  }
  img {
    width: 120px;
    height: 120px;
    margin: 0 auto;
    margin-top: 15px;
    margin-bottom: 7px;
  }
`
