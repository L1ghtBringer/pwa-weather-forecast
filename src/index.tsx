import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import backgroundDown from 'assets/backgroundDown.svg'
import backgroundDownLow from 'assets/backgroundDownLow.svg'
import backgroundUp from 'assets/backgroundUp.svg'
import backgroundUpLow from 'assets/backgroundUpLow.svg'
import { theme } from './styled'
import './font.css'

const Global = createGlobalStyle`
*,
::after,
::before{
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
body {
  overflow-x:hidden;
  margin-right: calc(-1 * (100vw - 100%));
  background: url(${backgroundUp}) top center no-repeat, url(${backgroundDown}) bottom center no-repeat #373af5;;
  background-size: 100%, 100%, cover;
  @media ${theme.media.phone}{
    background: url(${backgroundUpLow}) top center no-repeat, url(${backgroundDownLow}) bottom center no-repeat #373af5;
    background-size: 100%, 100%, cover;
  }
}
`

ReactDOM.render(
  <React.StrictMode>
    <ThemeProvider theme={theme}>
      <Global />
      <App />
    </ThemeProvider>
  </React.StrictMode>,
  document.getElementById('root')
)
