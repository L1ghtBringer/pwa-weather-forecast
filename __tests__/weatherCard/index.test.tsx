import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import WeatherCard from '../../src/components/WeatherCard'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../src/styled'

configure({ adapter: new Adapter() })

const data = {
    time: 1621454934, //19 May 2021
    temperature: 30,
    icon: '10d'
}

const shallowWithTheme = (children: any) => shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

describe('WeatherCard component test', () => {
  it('Should match the snapshot', () => {
    const weatherCard = shallowWithTheme(<WeatherCard {...data}/>)
    expect(weatherCard.html()).toMatchSnapshot()
    expect(weatherCard.exists()).toBe(true)
  })

  it('Test get date', () => {
    const weatherCard = shallow(<WeatherCard {...data}/>)
    expect(weatherCard.find("span").at(0).text()).toBe("19 May 2021");
  })

  it('Test temperature + ', () => {
    const weatherCard = shallow(<WeatherCard {...data}/>)
    expect(weatherCard.find("span").at(1).text()).toBe("+30°");
  })

  it('Test temperature - ', () => {
    const weatherCard = shallow(<WeatherCard {...data} temperature={-20}/>)
    expect(weatherCard.find("span").at(1).text()).toBe("-20°");
  })
})
