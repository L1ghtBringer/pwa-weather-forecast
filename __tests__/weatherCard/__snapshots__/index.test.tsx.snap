// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard component test Should match the snapshot 1`] = `"<div class=\\"sc-bdnxRM cEYTJY\\"><span>19 May 2021</span><img alt=\\"icons\\"/><span>+30Â°</span></div>"`;
